developers_chamber/scripts/qa.py

```diff
@@ -0,0 +1,246 @@
+    """
+    name = 'Check missing migrations'
+
+    def _run_check(self):
+        output = self._run_command('pydev make-migration --dry-run')
+        last_line = output.split('\n')[-1]
+        if last_line != 'No changes detected':
+            raise QAError('Found missing migration(s)!', output)
+
+
+class MigrationFilenamesQACheck(QACheck):
+    """
+    Checks that new migrations introduced in the branch have correct names.
+    """
+    name = 'Check migration filenames'
+
+    def _is_migration_file_with_wrong_name(self, path):
+        match = re.search(r'migrations\/([^\/]+)\.py$', path)
+        return bool(match and not re.search(r'^[0-9]{4}$', match.group(1)))
+
+    def _run_check(self):
+        wrong_name_files = []
+        for diff in self._get_diffs():
+            if diff.new_file and self._is_migration_file_with_wrong_name(diff.b_path):
+                wrong_name_files.append(diff.b_path)
+
+        if wrong_name_files:
+            raise QAError('Found wrongly named migration file:', '\n'.join(wrong_name_files))
+
+
+class MissingTranslationsQACheck(QACheck):
+    """
+    Checks that generating translations does not introduce any changes.
+    """
+    name = 'Check missing translations'
+
+    def _is_translation_file(self, path):
+        return bool(re.search(r'django\.po$', path))
+
+    def _run_check(self):
+        self._run_command('pydev make-msgs')
+
+        translation_files = []
+        for diff in self._get_unstaged():
+            if self._is_translation_file(diff.b_path) and re.findall(r'(\+|-)(msgstr|msgid)', diff.diff.decode()):
+                translation_files.append(diff.b_path)
+
+        if translation_files:
+            raise QAError('Found changes in following translation file(s):', '\n'.join(translation_files))
+
+
+class ImportOrderQACheck(QACheck):
+    """
+    Checks that isort applied on changed files does not introduce any changes.
+    """
+    name = 'Check import order'
+
+    def _is_python_file(self, path):
+        return bool(re.search(r'\.py$', path))
+
+    def _run_check(self):
+        changed_files = [diff.b_path for diff in self._get_diffs() if self._is_python_file(diff.b_path)]
+        if changed_files:
+            self._run_command('isort {}'.format(' '.join(changed_files)))
+        else:
+            return
+
+        wrong_import_order_files = set(changed_files) & set([diff.b_path for diff in self._get_unstaged()])
+        if wrong_import_order_files:
+            raise QAError('Found unsorted import(s) in following files:', '\n'.join(wrong_import_order_files))
+
+
+# CLI commands
+# ---------------------------------------------------------------------------------------------------------------------
+
+
+@cli.group()
+def qa():
+    """
+    Quality assurance commands.
+    """
+
+@qa.command()
+def all():
+    """
+    Runs all defined QA checks.
+    """
+    QACheckRunner(*(qa_check() for qa_check in QACheck.__subclasses__())).run()
```

Comment:
This will automatically run all checks defined (subclasses of `QATask`) without any firther efforts.

Reply:
That's kind of magical. What made you go this way instead of defining a list of QA checks to be run?

Reply:
It's up to discussion, but I like this solution, because it's one liner and you will never forget to "register" your check to be run.

Reply:
maybe register QA jobs in some method in the metamodel?

Reply:
I didn't want to use metaclass, that would be too obscure IMO. Personally I think this is OK, if you think it is not, I can sequentially list the classes here.

Reply:
explicit > implicit

Reply:
Not a golden rule, I think in this case it is more hassle than improvement, but OK.